bccsp/sw/fileks.go

```diff
@@ -355,27 +359,51 @@ func (ks *fileBasedKeyStore) loadKey(alias string) ([]byte, error) {
 func (ks *fileBasedKeyStore) createKeyStoreIfNotExists() error {
 	// Check keystore directory
 	ksPath := ks.path
-	missing, err := utils.DirMissingOrEmpty(ksPath)
 
-	if missing {
-		logger.Debugf("KeyStore path [%s] missing [%t]: [%s]", ksPath, missing, utils.ErrToString(err))
+	// Check if keystore directory exists
+	_, err := os.Stat(ksPath)
+	if os.IsNotExist(err) {
+		logger.Debugf("KeyStore path [%s] does not exist", ksPath)
 
-		err := ks.createKeyStore()
+		err = ks.createKeyStore()
 		if err != nil {
 			logger.Errorf("Failed creating KeyStore At [%s]: [%s]", ksPath, err.Error())
 			return nil
 		}
+	} else if err != nil {
+		return err
 	}
 
-	return nil
+	// Check if keystore directory is empty
```

Comment:
This is called **side effect** and **disinformation** as I specified in the coding guidelines here https://docs.google.com/document/d/1mU4S4U1dDpBk9AA180RMRB7g3OH887nqbcEPGD3px3Q/edit?usp=sharing 

- The name of the function is `CreateStoreIfNotExist()`
- What the function actually does? 
           1. creates the store if not exists
           2. creates the store if empty
- The function claims to do one thing but it does two things -- side effect and disinformation naming.

I understand that you didn't name it. Given that the function size is grown with additional details, I feel that the readability got impacted significantly.

I find the existing function more readable than this one. IMO, the existing code somewhat followed the formatting rule I specified in the above document but it used unjustifiable util. The existing function 
- checks whether the directory already exists or empty 
- if it so, it creates one 
It just took a few seconds to read the existing function.

 If possible, I would suggest fixing these issues.

Reply:
I ended up pulling the `dirEmpty` and `dirExists` functions up instead, and pushing the `CreateStoreIfNotExist` code up to the caller. I could have renamed the function `CreateStoreIfEmptyOrNotExist` but I think it was excessive and the code is clear what it is doing in the `Init` function.